flink-connector-dynamodb/src/test/java/org/apache/flink/connector/dynamodb/table/DynamoDbDynamicSinkITCase.java

```diff
@@ -0,0 +1,195 @@
+            new DynamoDbContainer(DockerImageName.parse(DockerImageVersions.DYNAMODB))
+                    .withCommand("-jar DynamoDBLocal.jar -inMemory -sharedDb")
+                    .withNetwork(Network.newNetwork())
+                    .withNetworkAliases("dynamodb");
+
+    @BeforeEach
+    public void setup() throws URISyntaxException {
+        testTableName = UUID.randomUUID().toString();
+        env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.setRestartStrategy(RestartStrategies.noRestart());
+        env.setParallelism(1);
+
+        dynamoDBHelpers = new DynamoDBHelpers(LOCALSTACK.getHostClient());
+    }
+
+    @Test
+    public void testSQLSink() throws ExecutionException, InterruptedException {
+
+        int expectedNumOfElements = 50;
+
+        dynamoDBHelpers.createTable(testTableName, PARTITION_KEY, SORT_KEY);
+        StreamTableEnvironment streamTableEnvironment =
+                StreamTableEnvironment.create(env, EnvironmentSettings.newInstance().build());
+
+        final String createTableStmt = getCreateTableStmt();
+        streamTableEnvironment.executeSql(createTableStmt);
+
+        final String datagenStmt = getDatagenStmt(expectedNumOfElements);
+        streamTableEnvironment.executeSql(datagenStmt);
+
+        final String insertSql = "INSERT INTO dynamo_db_table SELECT * from datagen;";
+        streamTableEnvironment.executeSql(insertSql).await();
+
+        Assertions.assertThat(dynamoDBHelpers.getItemsCount(testTableName))
+                .isEqualTo(expectedNumOfElements);
+    }
+
+    @Test
+    public void testTableAPISink() throws ExecutionException, InterruptedException {
+
+        int expectedNumOfElements = 50;
+
+        dynamoDBHelpers.createTable(testTableName, PARTITION_KEY, SORT_KEY);
+        StreamTableEnvironment streamTableEnvironment =
+                StreamTableEnvironment.create(env, EnvironmentSettings.newInstance().build());
+
+        final String createTableStmt = getCreateTableStmt();
+        streamTableEnvironment.executeSql(createTableStmt);
+
+        final String datagenStmt = getDatagenStmt(expectedNumOfElements);
+        streamTableEnvironment.executeSql(datagenStmt);
+
+        Table resultTable = streamTableEnvironment.sqlQuery("SELECT * FROM datagen");
+
+        resultTable.executeInsert("dynamo_db_table").await();
+
+        Assertions.assertThat(dynamoDBHelpers.getItemsCount(testTableName))
+                .isEqualTo(expectedNumOfElements);
+    }
+
+    private static String getDatagenStmt(int expectedNumOfElements) {
+        return "CREATE TEMPORARY TABLE datagen\n"
+                + "WITH (\n"
+                + "    'connector' = 'datagen',\n"
+                + "    'number-of-rows' = '"
+                + expectedNumOfElements
+                + "'\n"
+                + ")\n"
+                + "LIKE dynamo_db_table (EXCLUDING ALL);";
+    }
+
+    private static String getCreateTableStmt() {
```

Comment:
Q: can we move this to a java resource? same for the other statements

Reply:
+1 for resource usage